src/v/kafka/server/request_context.h

```diff
@@ -201,6 +202,12 @@ class request_context {
         return _conn->server().get_fetch_metadata_cache();
     }
 
+    bool is_topic_mutable(const model::topic& topic) const {
+        return _conn->server()
+          .cluster_link_frontend()
+          .is_topic_mutable_for_kafka_api(topic);
+    }
+
```

Comment:
nit: consider moving implementation to cc file so that the header doesn't have to include cluster_link frontend